packages/devtools_app/lib/src/service/editor/api_classes.dart

```diff
@@ -256,6 +289,196 @@ class ThemeChangedEvent extends EditorEvent {
   Map<String, Object?> toJson() => {Field.theme: theme};
 }
 
+/// An event sent by an editor when the current cursor position/s change.
+class ActiveLocationChangedEvent extends EditorEvent {
+  ActiveLocationChangedEvent({
+    required this.selections,
+    required this.textDocument,
+  });
+
+  ActiveLocationChangedEvent.fromJson(Map<String, Object?> map)
+    : this(
+        textDocument: TextDocument.fromJson(
+          map[Field.textDocument] as Map<String, Object?>,
+        ),
+        selections:
+            (map[Field.selections] as List<Object?>)
+                .cast<Map<String, Object?>>()
+                .map(EditorSelection.fromJson)
+                .toList(),
+      );
+
+  final List<EditorSelection> selections;
+  final TextDocument textDocument;
+
+  @override
+  EditorEventKind get kind => EditorEventKind.activeLocationChanged;
+
+  @override
+  Map<String, Object?> toJson() => {
+    Field.selections: selections,
+    Field.textDocument: textDocument,
+  };
+}
+
+/// A reference to a text document in the editor.
+class TextDocument with Serializable {
+  TextDocument({required this.uri, required this.version});
+
+  TextDocument.fromJson(Map<String, Object?> map)
+    : this(uri: map[Field.uri] as String, version: map[Field.version] as int);
+
+  final String uri;
+  final int version;
+
+  @override
+  Map<String, Object?> toJson() => {Field.uri: uri, Field.version: version};
+
+  @override
+  bool operator ==(Object other) {
+    return other is TextDocument &&
+        other.uri == uri &&
+        other.version == version;
+  }
+
+  @override
+  int get hashCode => Object.hash(uri, version);
+}
+
+// The starting and ending cursor positions in the editor.
+class EditorSelection with Serializable {
+  EditorSelection({required this.active, required this.anchor});
+
+  EditorSelection.fromJson(Map<String, Object?> map)
+    : this(
+        active: CursorPosition.fromJson(
+          map[Field.active] as Map<String, Object?>,
+        ),
+        anchor: CursorPosition.fromJson(
+          map[Field.anchor] as Map<String, Object?>,
+        ),
+      );
+
+  final CursorPosition active;
+  final CursorPosition anchor;
+
+  @override
+  Map<String, Object?> toJson() => {
+    Field.active: active.toJson(),
+    Field.anchor: anchor.toJson(),
+  };
+}
+
+/// Representation of a single cursor position in the editor.
+class CursorPosition with Serializable {
+  CursorPosition({required this.character, required this.line});
+
+  CursorPosition.fromJson(Map<String, Object?> map)
+    : this(
+        character: map[Field.character] as int,
+        line: map[Field.line] as int,
+      );
+
+  final int character;
```

Comment:
can you add dart doc? It's not obvious whether the cursor is before or after this character

Reply:
done!